src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -1453,6 +1528,30 @@ public sealed partial class ProxySupportAnnotation : IResourceAnnotation
         public required bool ProxyEnabled { get { throw null; } set { } }
     }
 
+    public sealed partial class PublishingCallbackAnnotation : IResourceAnnotation
+    {
+        public PublishingCallbackAnnotation(System.Func<PublishingContext, System.Threading.Tasks.Task> callback) { }
+
+        public System.Func<PublishingContext, System.Threading.Tasks.Task> Callback { get { throw null; } }
+    }
+
+    public sealed partial class PublishingContext
+    {
+        public PublishingContext(DistributedApplicationModel model, DistributedApplicationExecutionContext executionContext, System.IServiceProvider serviceProvider, Microsoft.Extensions.Logging.ILogger logger, System.Threading.CancellationToken cancellationToken, string outputPath) { }
+
+        public System.Threading.CancellationToken CancellationToken { get { throw null; } }
+
+        public DistributedApplicationExecutionContext ExecutionContext { get { throw null; } }
+
+        public Microsoft.Extensions.Logging.ILogger Logger { get { throw null; } }
```

Comment:
If this is the resource logger we should consider calling it ResourceLogger so it is clear.

Reply:
Do we also want to expose the publishing activity reporter here (or have APIs which proxy to it).

Reply:
I like ResourceLogger

Reply:
@mitchdenny are you making these updates

Reply:
This isn't the resource logger. It is a DI-injected `ILogger<Publisher>`. So it is "global" for the application, not per resource.

> Do we also want to expose the publishing activity reporter here (or have APIs which proxy to it).

The PublishingContext has an `IServiceProvider Services` (and a `DistributedApplicationExecutionContext`, which also has the `IServiceProvider`), which publishers can use to get the reporter. So I don't think we _need_ an API here, yet.

Reply:
Ah yes, my bad.